File: Container-With-Most-Water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int maxlen = 0;
        int left = 0;
        int right = n-1;
        while(left < right){
            maxlen = Math.max(maxlen, (right - left) * Math.min(height[left], height[right]));
            if(height[left] < height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return maxlen;
    }
}

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int i = 0, j = n - 1;
        int maxArea = 0;
        
        while (i < j) {
            int w = j - i;
            int h1 = height[i], h2 = height[j];
            int h = min(h1, h2);
            maxArea = max(maxArea, h * w);
            // Greedily moving Pointers
            h1 > h2 ? j-- : i++;
        }
        
        return maxArea;
    }
};